app/src/sharedTest/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentTest.kt

```diff
@@ -0,0 +1,427 @@
+        )
+      ).perform(click())
+
+      verifyFeatureFlagSwitchState(
+        0,
+        !downloadSupportFlag.currentValue
+      )
+    }
+  }
+
+  @Test
+  fun testFeatureFlagFragment_toggleDownloadSupportFlag_configChanges_valuePersists() {
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+      val downloadSupportFlag = getEphemeralFeatureFlags()[0]
+
+      scrollToPosition(0)
+      onView(
+        atPositionOnView(
+          R.id.feature_flags_recycler_view,
+          0,
+          R.id.feature_flag_switch
+        )
+      ).perform(click())
+
+      verifyFeatureFlagSwitchState(
+        0,
+        !downloadSupportFlag.currentValue
+      )
+
+      onView(isRoot()).perform(OrientationChangeAction.orientationLandscape())
+
+      verifyFeatureFlagSwitchState(
+        0,
+        !downloadSupportFlag.currentValue
+      )
+    }
+  }
+
+  private fun verifyFeatureFlagDisplayName(
+    position: Int,
+    expectedDisplayName: String
+  ) {
+    onView(
+      atPositionOnView(
+        R.id.feature_flags_recycler_view,
+        position,
+        R.id.feature_flag_label_text_view
+      )
+    ).check(
+      matches(
+        withText(expectedDisplayName)
+      )
+    )
+  }
+
+  private fun verifyFeatureFlagSyncStatus(
+    position: Int,
+    expectedSyncStatus: String
+  ) {
+    onView(
+      atPositionOnView(
+        R.id.feature_flags_recycler_view,
+        position,
+        R.id.sync_status_value_text_view
+      )
+    ).check(
+      matches(
+        withText(expectedSyncStatus)
+      )
+    )
+  }
+  private fun verifyFeatureFlagSwitchState(
+    position: Int,
+    expectedState: Boolean
+  ) {
+    onView(
+      atPositionOnView(
+        R.id.feature_flags_recycler_view,
+        position,
+        R.id.feature_flag_switch
+      )
+    ).check(matches(if (expectedState) isChecked() else not(isChecked())))
+  }
+  private fun getSyncStatusText(syncStatus: SyncStatus): String {
+    return when (syncStatus) {
+      SyncStatus.SYNC_STATUS_UNSPECIFIED ->
+        context.getString(R.string.feature_flag_unknown_sync_status)
+
+      SyncStatus.NOT_SYNCED_FROM_SERVER ->
+        context.getString(R.string.feature_flag_default_sync_status)
+
+      SyncStatus.SYNCED_FROM_SERVER ->
+        context.getString(R.string.feature_flag_server_sync_status)
+
+      else ->
+        context.getString(R.string.feature_flag_unknown_sync_status)
+    }
+  }
+  private fun scrollToPosition(position: Int) {
+    onView(withId(R.id.feature_flags_recycler_view)).perform(
+      scrollToPosition<RecyclerView.ViewHolder>(position)
+    )
+  }
+  private fun getEphemeralFeatureFlags(): List<EphemeralFeatureFlag> {
+    val provider = platformParameterControllerDebugImpl.loadEphemeralFeatureFlags()
+    return monitorFactory.waitForNextSuccessfulResult(provider)
+  }
+
+  private fun getFeatureFlagDisplayName(
+    id: FeatureFlagId
+  ): String {
```

Comment:
Ditto: Preferring single line since it is a single parameter declaration.
```suggestion
  private fun getFeatureFlagDisplayName(id: FeatureFlagId): String {
```

Reply:
uppdated